src/SAWScript/Crucible/LLVM/Override.hs

```diff
@@ -1938,6 +2101,172 @@ storePointsToValue opts cc env tyenv nameEnv base_mem maybe_cond ptr val maybe_i
+                        -- set, where n equals the size of the field
+                        -- (in bits) within the bitfield. Since `y` is
+                        -- 1-bit, this would make `fieldBitsBV` be
+                        -- 0b00000010, or 2.
+                        fieldBitsBV = BV.bit' bfWidth $
+                                      fromIntegral $ biFieldSize bfIndex
+
+                        -- A bitmask with the (n-1) least significant bits
+                        -- set to 1, where n equals the value of
+                        -- `fieldBitsBV`. For `y`, `leastBitsBV` would be
+                        -- equal to 0b00000001, or 1.
+                        leastBitsBV = BV.sub bfWidth fieldBitsBV (BV.one bfWidth)
+
+                        -- A bitmask with the bits corresponding to the
+                        -- field within the bitfield set to 1, and all
+                        -- other bits are set to 0. For `y`, `bitmaskBV`
+                        -- would be equal to 0b00001000, or 8.
+                        bitmaskBV = BV.shl bfWidth leastBitsBV $ fromIntegral bfOffset
+
+                        -- A bitmask with the bits corresponding to the
+                        -- field within the bitfield set to 0, and all
+                        -- other bits are set to 1. For `y`, `compBitmaskBV`
+                        -- would be equal to 0b11110111, or 247.
+                        compBitmaskBV = BV.complement bfWidth bitmaskBV
+                    compBitmaskSymBV <- W4.bvLit sym bfWidth compBitmaskBV
+
+                    -- Clear all of the bits in the bitfield
+                    -- corresponding to the field, leaving all other bits
+                    -- unchanged. If the original bitvector was 0b????????,
+                    -- then `bfBV'` is 0b????0???.
+                    bfBV' <- W4.bvAndBits sym bfBV compBitmaskSymBV
+
+                    -- Zero-extend the RHS value to be the same width as
+                    -- the bitfield. In our running example, we
+                    -- sign-extend 0b1 (1) to 0b00000001.
+                    rhsBV' <- W4.bvZext sym bfWidth rhsBV
+
+                    -- Left-shift the zero-extended RHS value such that
+                    -- its contents are in the same position as the field
+                    -- in the bitfield. In our running example, `rhsBV''`
+                    -- is 0b00001000, or 8.
+                    rhsBV'' <- W4.bvShl sym rhsBV' =<< W4.bvLit sym bfWidth bfOffsetBV
+
+                    -- Finally, set the bits in the bitfield to the RHS
+                    -- value with a bitwise-XOR operation. (We could just as
+                    -- well use bitwise-OR, but bitwise-XOR plays nicer with
+                    -- how What4 represents bitvectors). In our running
+                    -- example, `bfBV''` is 0b????1???.
+                    bfBV'' <- W4.bvXorBits sym bfBV' rhsBV''
```

Comment:
I think this can be  expressed more naturally in terms of `bvConcat` and `bvSelect`.

Reply:
I actually think the XOR formulation will be better, because the What4 AND/XOR semiring structure ought to  mean that subsequent updates to the same field will cancel, and update to disjoint fields will commute, etc.

Reply:
I don't have strong opinions on which approach is better _per se_, but I did try to implement your alternative approach in terms of `bvConcat` and `bvSelect`, and it wasn't entirely straightforward to do so. Unless I'm missing something, you have to have three cases:

1. The field occurs at the very LSBs of the bitfield. In that case, `bvSelect` the range from `rhsWidth` to `bfWidth - rhsWidth`, and then `bvConcat` that to the `rhsBV`.
2. The field occurs at the very MSBs of the bitfield. In that case, `bvSelect` the range from `0` to `bfFieldOffset`, and then `bvConcat` that to the `rhsBV`.
3. The field is in the middle of the bitfield, with other bits surrounding it on both sides. In that case, call `bvSelect` twice to obtain the surrounding bits, and then call `bvConcat` twice to concatenate these bitvectors to the `rhsBV`.

What's more, there's a fair bit of tiresome `NatRepr` manipulation needed to satisfy the inequalities demanded by `bvSelect`'s type signature. I've implemented it locally, and it is possible to do so, but the resulting code is not much more natural to look at than it is currently.